src/pages/studyView/table/StructuralVariantMultiSelectionTable.tsx

```diff
@@ -0,0 +1,782 @@
+        return _.chain(this.tableData)
+            .filter(data => this.flattenedFilters.includes(data.uniqueKey))
+            .sortBy<StructVarMultiSelectionTableRow>(data =>
+                ifNotDefined(order[data.uniqueKey], Number.POSITIVE_INFINITY)
+            )
+            .value();
+    }
+
+    @computed
+    get preSelectedRowsKeys() {
+        return this.preSelectedRows.map(row => row.uniqueKey);
+    }
+
+    @computed
+    get tableColumns() {
+        return this.props.columns.map(column => {
+            const columnDefinition = this.getDefaultColumnDefinition(
+                column.columnKey,
+                this.columnsWidth[column.columnKey],
+                this.cellMargin[column.columnKey]
+            );
+            if (column.columnTooltip) {
+                columnDefinition.tooltip = column.columnTooltip;
+            }
+            return columnDefinition;
+        });
+    }
+
+    @action.bound
+    toggleModal(panelName: string) {
+        this.modalSettings.modalOpen = !this.modalSettings.modalOpen;
+        if (!this.modalSettings.modalOpen) {
+            return;
+        }
+        this.modalSettings.modalPanelName = panelName;
+    }
+
+    @action.bound
+    closeModal() {
+        this.modalSettings.modalOpen = !this.modalSettings.modalOpen;
+    }
+
+    @autobind
+    toggleCancerGeneFilter(event: any) {
+        event.stopPropagation();
+        this.props.onChangeCancerGeneFilter(!this.props.filterByCancerGenes);
+    }
+
+    @computed get isFilteredByCancerGeneList() {
+        return (
+            !!this.props.cancerGeneFilterEnabled &&
+            this.props.filterByCancerGenes
+        );
+    }
+
+    @computed get allSelectedRowsKeysSet() {
+        return stringListToSet([
+            ...this.props.selectedRowsKeys,
+            ...this.preSelectedRowsKeys,
+        ]);
+    }
+
+    @autobind
+    isChecked(uniqueKey: string) {
+        return !!this.allSelectedRowsKeysSet[uniqueKey];
+    }
+
+    @autobind
+    isDisabled(uniqueKey: string) {
+        return _.some(this.preSelectedRowsKeys, key => key === uniqueKey);
+    }
+
+    @action.bound
+    onStructVarHover(rowId: string, isHovered: boolean) {
+        if (isHovered) {
+            this.hoveredStructVarTableRowIds.push(rowId);
+        } else {
+            _.pull(this.hoveredStructVarTableRowIds, rowId);
+        }
+    }
+
+    @action.bound
+    toggleSelectRow(uniqueKey: string) {
+        const record = _.find(
+            this.props.selectedRowsKeys,
+            key => key === uniqueKey
+        );
+        if (_.isUndefined(record)) {
+            this.props.onChangeSelectedRows(
+                this.props.selectedRowsKeys.concat([uniqueKey])
+            );
+        } else {
+            this.props.onChangeSelectedRows(
+                _.xorBy(this.props.selectedRowsKeys, [record])
+            );
+        }
+    }
+    @observable private _selectionType: SelectionOperatorEnum;
```

Comment:
i could be wrong but i thought you had to assign something here to get observable to work properly.

Reply:
This code is again directly copied from _MultiSelectionTable.tsx_. I think what you propose is not needed.